### PR TITLE
[17.0][MIG] hr_attendance_report_theoretical_time: Migration to version 17.0

### PR DESCRIPTION
**Pull request by juanjosesegui-tecnativa**
@Tecnativa TT50116

### CODE REVIEW

**Comment by juanjosesegui-tecnativa** [comment]
ping @pedrobaeza

File: hr_attendance_report_theoretical_time/tests/test_hr_attendance_report_theoretical_time.py
Comment (carolinafernandez-tecnativa) [inline hr_attendance_report_theoretical_time/tests/test_hr_attendance_report_theoretical_time.py]: Please use BaseCommon

File: hr_attendance_report_theoretical_time/views/hr_employee_views.xml
Comment (carolinafernandez-tecnativa) [inline hr_attendance_report_theoretical_time/views/hr_employee_views.xml:13]: active_ids is deprecated, making a warning on log, please check https://github.com/odoo/odoo/commit/daf05d48ac76d500aa1285184bdddc4c67641d58

**Review by carolinafernandez-tecnativa — COMMENTED** [review]
Functional review, LGTM.

Please squash administrative commits. 
Check https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate for details.

File: hr_attendance_report_theoretical_time/tests/test_hr_attendance_report_theoretical_time.py
Comment (carolinafernandez-tecnativa) [inline hr_attendance_report_theoretical_time/tests/test_hr_attendance_report_theoretical_time.py]: When using BaseCommon it is not necessary to add this.

File: hr_attendance_report_theoretical_time/views/hr_employee_views.xml
Comment (carolinafernandez-tecnativa) [inline hr_attendance_report_theoretical_time/views/hr_employee_views.xml]: @pedrobaeza i am not sure if this is correct.

**Review by carolinafernandez-tecnativa — COMMENTED** [review]
Squashing administrative is still pending, please check my comments before thanks :)

File: hr_attendance_report_theoretical_time/views/hr_employee_views.xml
Comment (juanjosesegui-tecnativa) [reply hr_attendance_report_theoretical_time/views/hr_employee_views.xml]: Ok, I will wait for Pedro's response to this change before pushing again with the squashing of the administrative commits.

Comment (pedrobaeza) [reply hr_attendance_report_theoretical_time/views/hr_employee_views.xml]: No, it isn't. It's a x2m field, so it requires another nomenclature. Maybe `[id]`, or `[(4, id)]`, but it should be tried.

File: hr_attendance_report_theoretical_time/tests/test_hr_attendance_report_theoretical_time.py
Comment (carolinafernandez-tecnativa) [inline hr_attendance_report_theoretical_time/tests/test_hr_attendance_report_theoretical_time.py]: This should be removed when using BaseCommon

**Review by carolinafernandez-tecnativa — APPROVED** [review]
LGTM Thanks :)
Please remember to squash administrative commits

File: hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report_views.xml
Comment (victoralmau) [inline hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report_views.xml]: I think this would not need to be changed.

File: hr_attendance_report_theoretical_time/views/hr_employee_views.xml
Comment (victoralmau) [inline hr_attendance_report_theoretical_time/views/hr_employee_views.xml]: Please check this. If I try to use this action from an employee there is an error.

**Review by victoralmau — CHANGES_REQUESTED** [review]

File: hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report_views.xml
Comment (juanjosesegui-tecnativa) [reply hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report_views.xml]: In the hr_attendance module, they have changed "report" to "reporting."

File: hr_attendance_report_theoretical_time/views/hr_employee_views.xml
Comment (juanjosesegui-tecnativa) [reply hr_attendance_report_theoretical_time/views/hr_employee_views.xml]: I can't find any errors.
If you test it again, please include a screenshot and I'll look into whether I can fix it. Thanks Victor

File: hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report_views.xml
Comment (pedrobaeza) [reply hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report_views.xml]: That has nothing to do with this one, as this is a new XML-ID not linked with the main one. Please respect existing XML-IDs.

File: hr_attendance_report_theoretical_time/tests/test_hr_attendance_report_theoretical_time.py
Comment (juanjosesegui-tecnativa) [reply hr_attendance_report_theoretical_time/tests/test_hr_attendance_report_theoretical_time.py]: All changes have been made and the administrative commits are squashed

**Comment by juanjosesegui-tecnativa** [comment]
All changes have been made and the administrative commits are squashed

ping @carolinafernandez-tecnativa

File: hr_attendance_report_theoretical_time/views/hr_employee_views.xml
Comment (victoralmau) [reply hr_attendance_report_theoretical_time/views/hr_employee_views.xml]: ejemplo

Comment (juanjosesegui-tecnativa) [reply hr_attendance_report_theoretical_time/views/hr_employee_views.xml]: Thank you, I was only able to get it to work by reverting to using active_ids, even though it will be deprecated in Odoo 18. Right now, [id] or [(4, id)] don't work.

**Review by victoralmau — APPROVED** [review]
Functional review OK

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**
